lib/index.js

```diff
@@ -1,9 +1,135 @@
 'use strict'
 
+const { normalizeOptions } = require('./util.js')
 const HttpAgent = require('./http.js')
 const HttpsAgent = require('./https.js')
 
+const AgentCache = new Map()
+
+const proxyEnv = {}
+for (const [key, value] of Object.entries(process.env)) {
+  const lowerKey = key.toLowerCase()
+  if (['https_proxy', 'http_proxy', 'proxy', 'no_proxy'].includes(lowerKey)) {
+    proxyEnv[lowerKey] = value
+  }
+}
```

Comment:
```suggestion
for (const key in process.env) {
  const lowerKey = key.toLowerCase()
  if (['https_proxy', 'http_proxy', 'proxy', 'no_proxy'].includes(lowerKey)) {
    proxyEnv[lowerKey] = process.env[key]
  }
}
```

Reply:
Sadly, `Object.entries` does not return an iterator.

Reply:
This will "somewhat" nondeterministically clobber entries w/ duplicated content based on casing and I think we should acknowledge that here in this review and say "Yeah we don't support that"